app/controllers/api/v2/compute_resources_controller.rb

```diff
@@ -247,12 +247,14 @@ def available_virtual_machines
       param :id, :identifier, :required => true
       param :vm_id, :identifier, :required => true
       def show_vm
-        @vm = @compute_resource.find_vm_by_uuid(params[:vm_id])
+        begin
+          @vm = @compute_resource.find_vm_by_uuid(params[:vm_id])
+        rescue
+          raise ::Foreman::Exception.new(N_("Virtual machine was not found by id %{vm_id}") % {:vm_id => params[:vm_id]})
+        end
         attributes = @vm.attributes.deep_symbolize_keys
-        attributes.delete(:parent)
-        render :json => attributes
-      rescue
-        raise ::Foreman::Exception.new(N_("Virtual machine was not found by id %{vm_id}") % {:vm_id => params[:vm_id]})
+        attributes[:provider] = @compute_resource.provider
+        render :json => attributes.as_json(:except => [:label_fingerprint, :fingerprint, :parent])
```

Comment:
`:label_fingerprint` and `:fingerprint` were originally not excluded. What are the reasons for not rendering them?

Reply:
The as_json function does not work and therefore the rendering failed when trying to render label_fingerprint and fingerprint. This failed because there were Non-printable characters in those fields which appear only in GCE.
This is the error when rendering the JSON without excluding these fields:
Encoding::UndefinedConversionError: "\xE3" from ASCII-8BIT to UTF-8

Reply:
I was just curious, seems like excluding them is the easiest option.